src/loader.js

```diff
@@ -146,15 +147,186 @@
+  function RegistryPrototype() {}
+
+  // 4.4.1
+  Registry.prototype.constructor = Registry;
+
+  // 4.4.2 is inlined in 4.1.1
+
+  // 4.4.3
+  RegistryPrototype.prototype.entries = function() {
+    if (typeof this !== 'object')
+      throw new TypeError('cannot get entries of a non-registry');
+    if (__global.Map) {
+      // native iterator
+      return this._registry.registryMap.entries();
+    } else if (__global.Symbol && __global.Symbol.iterator) {
+      // polyfilled iterator
+      var keys = Object.keys(this._registry.registryMap);
+      var keyIndex = 0;
+      var instance = this;
+      return {
+        next: function() {
+          if (keyIndex < keys.length) {
+            return {
+              value: [keys[keyIndex], instance._registry.registryMap[keys[keyIndex++]]],
+              done: false
+            };
+          } else {
+            return {
+              value: undefined,
+              done: true
+            };
+          }
+        }
+      };
+    } else {
+      throw new Error('Symbol.iterator must be polyfilled to iterate over registry entries');
+    }
+  }
+
+  // 4.4.4
+  RegistryPrototype.prototype.keys = function() {
+    if (typeof this !== 'object')
+      throw new TypeError('invalid registry');
+    if (__global.Map) {
+      // native iterator
+      return this._registry.registryMap.keys();
+    } else if (__global.Symbol && __global.Symbol.iterator) {
+      // polyfilled iterator
+      var keys = Object.keys(this._registry.registryMap);
+      var keyIndex = 0;
+      return {
+        next: function() {
+          if (keyIndex < keys.length) {
+            return {
+              value: keys[keyIndex++],
+              done: false
+            };
+          } else {
+            return {
+              value: undefined,
+              done: true
+            };
+          }
+        }
+      };
+    } else {
+      throw new Error("Symbol.iterator must be polyfilled in order to iterate over registry keys")
+    }
+  }
+
+  // 4.4.5
+  RegistryPrototype.prototype.values = function() {
+    if (typeof this !== 'object')
+      throw new TypeError('invalid registry');
+    if (__global.Map) {
+      // native iterator
+      return this._registry.registryMap.values();
+    } else if (__global.Symbol && __global.Symbol.iterator) {
+      // polyfilled iterator
+      var keys = Object.keys(this._registry.registryMap);
+      var keyIndex = 0;
+      var instance = this;
+      return {
+        next: function() {
+          if (keyIndex < keys.length) {
+            return {
+              value: instance._registry.registryMap[keys[keyIndex++]],
+              done: false
+            };
+          } else {
+            return {
+              value: undefined,
+              done: true
+            };
+          }
+        }
+      };
+    } else {
+      throw new Error("Symbol.iterator must be polyfilled in order to iterate over registry values");
+    }
+  }
```

Comment:
Probably a good idea -- especially if we need to use Maps elsewhere. Will change.